Mathlib/Topology/MetricSpace/CoveringNumbers.lean

```diff
@@ -243,4 +390,33 @@ theorem packingNumber_two_mul_le_externalCoveringNumber (ε : ℝ≥0) (A : Set
       · exact hf' x
       · simpa [edist_comm, hxy] using hf' y
 
+theorem coveringNumber_le_packingNumber (ε : ℝ≥0) (A : Set X) :
+    coveringNumber ε A ≤ packingNumber ε A := by
+  by_cases h_top : packingNumber ε A ≠ ⊤
+  · rw [← card_maximalSeparatedSet h_top]
+    refine (iInf_le _ (maximalSeparatedSet ε A)).trans (le_of_eq ?_)
+    simp [maximalSeparatedSet_subset, iInf_pos, isCover_maximalSeparatedSet h_top]
+  · simp only [ne_eq, Decidable.not_not] at h_top
+    simp [h_top]
+
+theorem coveringNumber_two_mul_le_externalCoveringNumber (ε : ℝ≥0) (A : Set X) :
+    coveringNumber (2 * ε) A ≤ externalCoveringNumber ε A := by
+  rcases Set.eq_empty_or_nonempty A with (h_empty | h_nonempty)
+  · simp [h_empty]
+  refine (coveringNumber_le_packingNumber _ A).trans ?_
+  exact packingNumber_two_mul_le_externalCoveringNumber ε A
+
+lemma coveringNumber_subset_le (h : A ⊆ B) :
+    coveringNumber ε A ≤ coveringNumber (ε / 2) B := by
+  calc coveringNumber ε A
```

Comment:
otherwise style guidelines would require to indent all lines below the first `calc` line.
```suggestion
    coveringNumber ε A ≤ coveringNumber (ε / 2) B := calc
  coveringNumber ε A
```